pkg/scheduler/framework/plugins/noderesources/test_util.go

```diff
@@ -22,6 +22,83 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+type containerResource struct {
+	cpu    int
+	memory int
+}
+
+func makePod(node string, milliCPU, memory int64) *v1.Pod {
+	return &v1.Pod{
+		Spec: v1.PodSpec{
+			NodeName: node,
+			Containers: []v1.Container{
+				{
+					Resources: v1.ResourceRequirements{
+						Requests: v1.ResourceList{
+							v1.ResourceCPU:    *resource.NewMilliQuantity(milliCPU, resource.DecimalSI),
+							v1.ResourceMemory: *resource.NewQuantity(memory, resource.DecimalSI),
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func makePodWithContainers(node string, resources ...containerResource) *v1.Pod {
+	if len(resources) == 0 {
+		return &v1.Pod{
+			Spec: v1.PodSpec{
+				NodeName:   node,
+				Containers: []v1.Container{{}},
+			},
+		}
+	}
+
+	var containers []v1.Container
+	for i := range resources {
+		cpu := resources[i].cpu
+		memory := resources[i].memory
+		containers = append(containers, v1.Container{
+			Resources: v1.ResourceRequirements{
+				Requests: v1.ResourceList{
+					v1.ResourceCPU:    *resource.NewMilliQuantity(int64(cpu), resource.DecimalSI),
+					v1.ResourceMemory: *resource.NewQuantity(int64(memory), resource.DecimalSI),
+				},
+			},
+		})
+	}
+
+	return &v1.Pod{
+		Spec: v1.PodSpec{
+			NodeName:   node,
+			Containers: containers,
+		},
+	}
+}
+
+func makePodWithExtendedResource(node string, milliCPU, memory int64, extendedResource map[string]int64) *v1.Pod {
+	resourceList := make(map[v1.ResourceName]resource.Quantity)
```

Comment:
Don't quite need to differentiate extended res from core res - this can be achieved simply by `st.MakePod().Node().Req()` - no matter it's extended resource or not.